Polish D2k weapons and add wall explosion

reaperrr opened the pull request:
Closes #12766.
Closes #12768.

Note: The missile tank and missile turret weapons didn't have explosion sounds in the original, either, but I left them enabled here because... well, there would be almost no weapon explosion sounds otherwise.

Adding to Next milestone because it's fairly uncontroversial polish and easy to review/test.

reaperrr committed:
Fix explosion sound, speed and angle of D2k grenade

reaperrr committed:
Only play explosion sound for Missile Tank and Tower in D2k

But not on Bazooka and Quad's Rocket.

reaperrr committed:
Make D2k large missile speeds consistent

reaperrr committed:
Make Quad Rocket speed a multiple of 32

reaperrr committed:
Remove explosion sound from Fremen weapon

reaperrr committed:
Set up D2k wall explosion

ltem commented:
:+1: 

(I still don't know what the Fremen gun sound shall resemble ^^)

pchote commented on mods/d2k/sequences/misc.yaml:
```diff
@@ -52,6 +52,11 @@ explosion:
+	wall_explosion: DATA.R8
+		Start: 4130
+		Length: 8
+		Tick: 200
```
This feels too slow.  120 is better.

pchote commented on mods/d2k/rules/structures.yaml:
```diff
@@ -649,6 +649,9 @@ wall:
+	Explodes:
+		Weapon: WallExplode
+		EmptyWeapon: WallExplode
 	ThrowsShrapnel:
```
I didn't notice shrapnel when I tested the original, nor does it make sense (concrete walls don't contain anything combustable).  Can you remove this?

reaperrr replied:
I played the original yesterday and they definitely spawned up to 2 debris.

pchote replied:
Ah ok, fair enough.

reaperrr replied:
Did some more testing and they spawned *exactly* 2 debris, 1 small and 1 large.
I noticed that our 'Pieces' represent the number for the respective list entry rather than min/max total, so I changed it to `1, 1`. Other buildings will need that fixed as well I guess, but not in this PR.

reaperrr committed:
Fix BlendMode of previously unused D2k trails

reaperrr committed:
Tweak D2k debris

- Enabled shadow
- reduced maximum speed
- considerably increased 'launch angle'
- increased bounce count to 3 and toned down BounceRangeModifier
- reduced TrailInterval to 1 for Debris2
- removed explosion sounds to a) match original and b) make them less annoying
- fixed debris explosion animations to match original
- used different trail anim for Debris2 and Debris3

reaperrr committed:
Fix D2k Wall debris count and types

Two debris, of which one is large, matches the original.
Also made max range match weapon range.

reaperrr committed:
Fix D2k wall armor type

TibEd said 'none', but that can't be true, as in the original anti-tank weapons are far more effective versus walls than machine guns, so I decided to just use the wall armor type.

reaperrr commented:
Updated, fine-tuned debris and fixed wall armor type.

GraionDilach reviewed (approved):
:+1:

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c33ae2b6c1823075a7099ff8782b20d016064ad8
